pkg/cmd/subcommand/demo.go

```diff
@@ -0,0 +1,258 @@
+
+func init() {
+	w = os.Stdout
+	availableVersions = []string{"v0.3.1", "v0.4.0", "latest"}
+}
+
+// NewDemo returns an initialized Init instance
+func NewDemo(ioStreams genericclioptions.IOStreams) *Demo {
+	return &Demo{
+		IOStreams: ioStreams,
+	}
+}
+
+func NewCmdDemo(restClient util.Getter, ioStreams genericclioptions.IOStreams) *cobra.Command {
+
+	i := NewDemo(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:                   "demo <args>...",
+		DisableFlagsInUseLine: true,
+		Short:                 "Create OpenFunction demo.",
+		Long: `
+         You can use ofn demo to run the OpenFunction demo.
+         Available options are: --delete, --region-cn, --verbose
+`,
+		Example: "ofn demo",
+		Run: func(cmd *cobra.Command, args []string) {
+			if i.Delete {
+				util.CheckErr(i.DeleteKind(cmd, args))
+			} else {
+				util.CheckErr(i.RunKind(cmd))
+				installCMD := NewCmdInstall(restClient, ioStreams)
+				installCMD.SetArgs([]string{
+					fmt.Sprintf("install"),
+					fmt.Sprintf("--all"),
+					fmt.Sprintf("--region-cn=%v", i.RegionCN),
+					fmt.Sprintf("--version=v0.4.0"),
+				   fmt.Sprintf("--timeout=%v",10*time.Minute)},
+				)
+				installCMD.Execute()
+				util.CheckErr(i.RunDemo(cmd))
+			}
+		},
+	}
+
+	cmd.Flags().BoolVar(&i.Verbose, "verbose", false, "Show verbose information.")
+	cmd.Flags().BoolVar(&i.RegionCN, "region-cn", false, "For users who have limited access to gcr.io or github.com.")
+	cmd.Flags().BoolVar(&i.Delete, "delete", false, "Used to specify the version of OpenFunction to be uninstalled.")
+	cmd.Flags().DurationVar(&i.Timeout, "timeout", 5*time.Minute, "Set timeout time. Default is 5 minutes.")
+	// In order to avoid too many options causing misunderstandings among users,
+	// we have hidden the following parameters,
+	// but you can still find their usage instructions in the documentation.
+	cmd.Flags().MarkHidden( "version")
+	cmd.Flags().MarkHidden("ingress")
+	cmd.Flags().MarkHidden("cert-manager")
+	cmd.Flags().MarkHidden("shipwright")
+	cmd.Flags().MarkHidden("keda")
+	cmd.Flags().MarkHidden("dapr")
+	return cmd
+}
+
+
+func (i *Demo) RunKind(cmd *cobra.Command) error {
+	operator := common.NewOperator(runtime.GOOS, i.OpenFunctionVersion, i.Timeout, i.RegionCN, i.Verbose)
+	ti := util.NewTaskInformer("Demo")
+	Continue := func() bool {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Fprintln(w, ti.BeforeTask("A Kind cluster will be created and the OpenFunction Demo will be launched in it...\n"+
+			"Enter 'y' to continue and 'n' to abort:"))
+
+		for {
+			fmt.Fprint(w, ti.BeforeTask("-> "))
+			text, _ := reader.ReadString('\n')
+			// convert CRLF to LF
+			text = strings.Replace(text, "\n", "", -1)
+
+			if strings.Compare("y", text) == 0 {
+				return true
+			}
+			if strings.Compare("n", text) == 0 {
+				return false
+			}
+		}
+	}
+
+	ctx, done := context.WithTimeout(
+		context.Background(),
+		i.Timeout,
+	)
+	defer done()
```

Comment:
need to delete kind cluster in the defer func

Reply:
Maybe should delete the kind cluster after running RunDemo, not after installing kind by `RunKind`.
If we add the kind cluster auto-deleting/release mechanism, maybe the `--delete` option is not needed.
A user just need to use run `ofn demo xxx` to figure out how of works, and needn't care about the demo env setup or cleanup?

Reply:
Yes, we have found the right place to insert the logic for deleting the kind of cluster that failed to be created.
We should perform a rollback operation (i.e. delete similar clusters) after an installation failure of OpenFunction to ensure that everything is back to normal the next time the user executes the `ofn demo` command.